Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/transform/CoreX12AuditDataTransform.java

```diff
@@ -0,0 +1,467 @@
+import java.net.UnknownHostException;
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+import org.apache.log4j.Logger;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmission;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmissionResponse;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeRequest;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeResponse;
+
+/**
+ * Class provides generic Audit Data Transformation APIs utilized across all the services
+ *
+ * @author nsubrama
+ */
+public class CoreX12AuditDataTransform {
+
+    private static final XDRTransforms oXDR = new XDRTransforms();
+
+    private static final Logger LOG = Logger.getLogger(CoreX12AuditDataTransform.class);
+
+    public static AuditSourceIdentificationType getAuditSourceIdentificationType() {
+        String hcid = HomeCommunityMap.getLocalHomeCommunityId();
+        String homeCommunityName = HomeCommunityMap.getHomeCommunityName(hcid);
+        return AuditDataTransformHelper.createAuditSourceIdentification(hcid, homeCommunityName);
+    }
+
+    protected static AuditMessageType.ActiveParticipant getActiveParticipant(UserType oUserInfo) {
+        // Create Active Participant Section
+        // create a method to call the AuditDataTransformHelper - one expectation
+        AuditMessageType.ActiveParticipant participant = createActiveParticipantFromUser(
+            oUserInfo, Boolean.TRUE);
+        if (null != oUserInfo.getRoleCoded()) {
+            participant.getRoleIDCode().add(AuditDataTransformHelper.createCodeValueType(oUserInfo.getRoleCoded().getCode(), "",
+                oUserInfo.getRoleCoded().getCodeSystemName(), oUserInfo.getRoleCoded().getDisplayName()));
+        }
+        return participant;
+    }
+
+    /**
+     * Create the <code>AuditMessageType.ActiveParticipant</code> for an audit log record.
+     *
+     * @param userInfo
+     * @param userIsReq
+     * @return <code>AuditMessageType.ActiveParticipant</code>
+     */
+    private static AuditMessageType.ActiveParticipant createActiveParticipantFromUser(UserType userInfo,
+        Boolean userIsReq) {
+        String ipAddr = null;
+        AuditMessageType.ActiveParticipant participant = new AuditMessageType.ActiveParticipant();
+
+        if (ipAddr == null) {
+            try {
+                ipAddr = InetAddress.getLocalHost().getHostAddress();
+            } catch (UnknownHostException ex) {
+                LOG.error("UnknownHostException thrown getting local host address.", ex);
+                return null;
+            }
+        }
+
+        // Set the User Id
+        String userId = null;
+        if (userInfo != null && userInfo.getUserName() != null && userInfo.getUserName().length() > 0) {
+            userId = userInfo.getUserName();
+        }
+
+        if (userId != null) {
+            participant.setUserID(userId);
```

Comment:
why are we doing this here, instead of inside the previous if statement.  unnecessary if statement.